devito/passes/iet/errors.py

```diff
@@ -0,0 +1,87 @@
+import cgen as c
+import numpy as np
+from sympy import Not
+
+from devito.ir.iet import (Call, Conditional, EntryFunction, Iteration, List,
+                           Return, FindNodes, FindSymbols, Transformer,
+                           make_callable)
+from devito.passes.iet.engine import iet_pass
+from devito.symbolics import CondEq, DefFunction
+from devito.types import Eq, Inc, Symbol
+
+__all__ = ['check_stability', 'error_mapper']
+
+
+def check_stability(graph, options=None, rcompile=None, sregistry=None, **kwargs):
+    """
+    Check if the simulation is stable. If not, return to Python as quickly as
+    possible with an error code.
+    """
+    if options['errctl'] != 'max':
+        return
+
+    _, wmovs = graph.data_movs
+
+    _check_stability(graph, wmovs=wmovs, rcompile=rcompile, sregistry=sregistry)
+
+
+@iet_pass
+def _check_stability(iet, wmovs=(), rcompile=None, sregistry=None):
+    if not isinstance(iet, EntryFunction):
+        return iet, {}
+
+    # NOTE: Stability is a domain-specific concept, hence looking for time
+    # Iterations and TimeFunctions is acceptable
+    efuncs = []
+    includes = []
+    mapper = {}
+    for n in FindNodes(Iteration).visit(iet):
+        if not n.dim.is_Time:
+            continue
+
+        functions = [f for f in FindSymbols().visit(n)
```

Comment:
Nitpicking: Might be cases where there is no stepping dim and only time dime, but can be for later if someone runs into it (unlikely).